### ISSUE TITLE
Purescript-0.11.7 fails to compile

### ISSUE THREAD

**Issue by ravloony** [NixOS/nixpkgs#32138]
## Issue description

Purescript-0.11.7 fails to compile, with the following output:

```
Preprocessing executable 'purs' for purescript-0.11.7...
[ 1 of 14] Compiling Paths_purescript ( dist/build/autogen/Paths_purescript.hs, dist/build/purs/purs-tmp/Paths_purescript.o )
[ 2 of 14] Compiling Version          ( app/Version.hs, dist/build/purs/purs-tmp/Version.o )
[ 3 of 14] Compiling Command.REPL     ( app/Command/REPL.hs, dist/build/purs/purs-tmp/Command/REPL.o )
[ 4 of 14] Compiling Command.Publish  ( app/Command/Publish.hs, dist/build/purs/purs-tmp/Command/Publish.o )
[ 5 of 14] Compiling Command.Ide      ( app/Command/Ide.hs, dist/build/purs/purs-tmp/Command/Ide.o )

app/Command/Ide.hs:144:30: error:
    • Couldn't match type ‘[Char]’ with ‘Text’
      Expected type: Opts.Parser Text
        Actual type: Opts.Parser GHC.Base.String
    • In the second argument of ‘(<$>)’, namely
        ‘Opts.strOption
           (Opts.long "log-level" `mappend` Opts.value ""
            `mappend`
              Opts.help "One of \"debug\", \"perf\", \"all\" or \"none\"")’
      In the second argument of ‘(<*>)’, namely
        ‘(parseLogLevel
          <$>
            Opts.strOption
              (Opts.long "log-level" `mappend` Opts.value ""
               `mappend`
                 Opts.help "One of \"debug\", \"perf\", \"all\" or \"none\""))’
      In the first argument of ‘(<*>)’, namely
        ‘ServerOptions
         <$>
           optional
             (Opts.strOption (Opts.long "directory" `mappend` Opts.short 'd'))
         <*> many (Opts.argument Opts.str (Opts.metavar "Source GLOBS..."))
         <*>
           Opts.strOption
             (Opts.long "output-directory" `mappend` Opts.value "output/")
         <*>
           (fromIntegral
            <$>
              Opts.option
                Opts.auto
                (Opts.long "port" `mappend` Opts.short 'p'
                 `mappend` Opts.value (4242 :: Integer)))
         <*> Opts.switch (Opts.long "no-watch")
         <*> flipIfWindows (Opts.switch (Opts.long "polling"))
         <*>
           (parseLogLevel
            <$>
              Opts.strOption
                (Opts.long "log-level" `mappend` Opts.value ""
                 `mappend`
                   Opts.help "One of \"debug\", \"perf\", \"all\" or \"none\""))’
```

### Steps to reproduce

Install nixos from master.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.13.11, NixOS, 18.03.git.9e241d0 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-17.09pre113138.96457d26dd"`
 - channels(ravloony): `"nixos-17.09pre105622.f0fac3b578"`
 - nixpkgs: `/etc/nixos/nixpkgs`

**Comment by joncfoo** [comment]
I believe this is failing since the `purescript` haskell package now has a dependency on `optparse-applicative-0.14.0.0`.  The build error is due to it trying to build with `optparse-applicative-0.13.2.0` which is the default version in nixpkgs.

I'd be interested in helping to fix this but am not sure where to override its dependent package.

**Comment by ravloony** [comment]
Fixed in #32181
